provider/gandi/gandi.go

```diff
@@ -48,15 +49,49 @@ type GandiProvider struct {
 	LiveDNSClient LiveDNSClientAdapter
 	DomainClient  DomainClientAdapter
 	domainFilter  endpoint.DomainFilter
+	PreferALIAS   bool
 	DryRun        bool
+	TXTPrefix     string
+	TXTSuffix     string
 }
 
-func NewGandiProvider(ctx context.Context, domainFilter endpoint.DomainFilter, dryRun bool) (*GandiProvider, error) {
+const (
+	// RecordTypeALIAS is a RecordType enum value
+	RecordTypeALIAS = "ALIAS"
+)
+
+// inferZone determines the zone based on the RrsetName
+func inferZone(RrsetName string, TXTPrefix string, TXTSuffix string) string {
+	cleanRrsetName := strings.Replace(RrsetName, TXTPrefix, "", 1)
+	cleanRrsetName = strings.Replace(cleanRrsetName, TXTSuffix, "", 1)
+	cleanRrsetName = strings.Replace(cleanRrsetName, "cname-", "", 1)
```

Comment:
This just screams layering violation to me. Why is the provider mucking about with things that are in the purview of the TXT registry? Why is this only removing a prefix of `cname-`?

Reply:
There probably is a better way to do it but not sure how - the issue is that gandi's API requires us explicitly declare which "zone" the records we are manipulating are in.
When working on a CNAME on an apex domain, TXTRegistry creates the old and new txt records, with and without the type prefix. When we receive the endpoint with value `cname-example.com 0 IN TXT  \"heritage=external-dns,external-dns/owner=default\" []`, we have no way to reliability find the zone without this kind of tomfoolery, do we?

Reply:
I have tried to improve this with 83a1fcd014d4d9c5b640e305afc29b708f72caba - I still feel like we have no better way to compute a "zone" from an `Endpoint`

We can spin this change in a new PR if you think it is necessary?